### DISCUSSION TITLE
Multi-agent reinforcement learning in BlueSky

### DISCUSSION THREAD

**Discussion by jooste** [TUDelft-CNS-ATM/bluesky#253] [Ideas]
Drop by for all your discussions on reinforcement learning applications in BlueSky! Let's start with some links to previous work:

[Bart Vonk's thesis work](http://resolver.tudelft.nl/uuid:2e776b60-cd4e-4268-93e3-3fcc81cd794f) ([BlueSky fork](https://github.com/BVonk/bluesky))

[Dennis van der Hoff thesis work](http://resolver.tudelft.nl/uuid:4d02e51a-0187-404d-b465-7ae01feba8e8) ([BlueSky fork](https://github.com/devanderhoff/bluesky))

Marc Brittain's PhD work [1](https://www.researchgate.net/publication/338084245_One_to_Any_Distributed_Conflict_Resolution_with_Deep_Multi-Agent_Reinforcement_Learning_and_Long_Short-Term_Memory), [2](https://arxiv.org/abs/1905.01303) ([BlueSky fork](https://github.com/marcbrittain/bluesky))

Marta Ribeiro's PhD work [1](https://www.researchgate.net/publication/346647401_Determining_Optimal_Conflict_Avoidance_Manoeuvres_At_High_Densities_With_Reinforcement_Learning) ([BlueSky fork](https://github.com/MJRibeiroTUDelft/bluesky))

**Comment by LuisNavarroGarcia** [comment]
Hi!

I think this this is the right place to post about it, but if preferred in some other place I will delete it from here.

I've been working on the topic for my bachelor's final year thesis, when trying to visualize trajectories I open a client and I connect it to a running server (The GUI at start-up shows a pop-up window with the available port, that corresponds to the running server and I just click on that). The simulation is being run with an environment by RLlib but the **client doesn't seem to display anything** related to the simulation so I don't really know if I'm missing something. Any clarification would be of great help as after several tries I've run out of ideas of what I am doing wrong.

I've been looking at Dennis van der Hoff's work and it has been of great help so, thank you very much for sharing!

Regards,

Luis.

**Reply by jooste** [reply]
Hi Luis,

Welcome to the bluesky discussion board! The problem you describe is difficult to investigate remotely, since there isn't a traceback or other visual error to analyse. If you are certain that you are connecting to the correct bluesky simulation with your client, the first thing I would check is to put some print statements on the sim side and on the gui side to see:
- If the simulation is really running, and if it contains aircraft
- If the simulation is sending data
- If the client is receiving data at all
Hopefully that reveals something we can fix.

**Reply by LuisNavarroGarcia** [reply]
Thank you for the response!

¡I will try my best! In case I find something I'll open a new issue but it is most likely my own fault.

Regards,

Luis.
